openfe/protocols/openmm_utils/system_creation.py

```diff
@@ -235,3 +254,234 @@ def _add_small_mol(comp,
+
+    Parameters
+    ----------
+    offmol: openff.toolkit.Molecule
+      The Molecule to assign AM1BCC charges for.
+    charge_backend : str
+      Which backend to use to generate charges. Available options are:
+
+      * `ambertools`: Use AmberTools' antechamber to generate charges.
+      * `oechem`: Use the OpenEye toolkit to generate charges.
+    """
+    backend = _get_toolkit_wrapper_charge_backend(charge_backend)
+
+    try:
+        offmol.assign_partial_charges(
+            partial_charge_method='am1bcc',
+            use_conformers=conformer,
+            toolkit_registry=backend(),
+        )
+    # if we get a confgen failure fallback to using the existing conformer
+    except ValueError:
+        offmol.assign_partial_charges(
+            partial_charge_method='am1bcc',
+            use_conformers=offmol.conformers,
+            toolkit_registry=backend(),
+        )
+
+
+def assign_nagl_am1bcc_charges(
+    offmol: OFFMol,
+    nagl_model: Optional[str] = "openff-gnn-am1bcc-0.0.1-rc.1.pt"
+) -> None:
+    """
+    Assign partial charges using the NAGL ML Model.
+
+    Parameters
+    ----------
+    offmol: openff.toolkit.Molecule
+      The Molecule to assign AM1BCC charges for.
+    nagl_model : Optional[str]
+      The nagl model to use for partial charge assignment.
+      If None, will use latest available model.
+    """
+    if not HAS_NAGL:
+        errmsg = ("The NAGLToolkitWrapper is not available, you may be using "
+                  "and older version of the OpenFF toolkit - you need 0.14.4 "
+                  "or above.")
+        raise ImportError(errmsg)
+
+    if nagl_model is None:
+        # It's not fully clear that the models will always be sort ordered
+        # see: https://github.com/openforcefield/openff-nagl-models/issues/12
+        from openff.nagl_models import list_available_nagl_models
+        nagl_model = list_available_nagl_models()[-1]
+
+    offmol.assign_partial_charges(
+        partial_charge_method=nagl_model,
+        toolkit_registry=NAGLToolkitWrapper(),
+    )
+
+
+def assign_am1bccelf10_charges(
+    offmol: OFFMol,
+    charge_backend: str = 'ambertools',
+    ambertools_generate_n_conformers: int = 500,
+) -> None:
+    """
+    Assign AM1BCC charges using a given toolkit.
+
+    Parameters
+    ----------
+    offmol : openff.toolkit.Molecule
+      The Molecule to assign AM1BCCELF10 charges for.
+    charge_backend : str
+      Which backend to use to generate charges. Available options are:
+
+      * `ambertools`: Use AmberTools' antechamber to generate charges.
+      * `oechem`: Use the OpenEye toolkit to generate charges.
+
+    ambertools_generate_n_conformers : int
+      The number of conformers to initially generate for elf10 selection.
+      Note: this is only used by the `ambertools` backend option.
+    """
+    backend = _get_toolkit_wrapper_charge_backend(charge_backend)
+
+    if charge_backend.lower() == 'oechem':
+        offmol.assign_partial_charges(
```

Comment:
This might be an issue with the "without oechem" method we are using - more specifically this https://github.com/openforcefield/openff-toolkit/blob/dcd78c97a161b522665d7a4c74ea1eefc5ec2ceb/openff/toolkit/utils/openeye_wrapper.py#L2420 might be a problem because it might attempt to run with rdkit instead of oechem?

Reply:
I imagine this will replace that hack

Reply:
unfortunately those are two seprate things - if I remember correctly wasn't the dissallowed backends necessary because of the call to smiles serialisation when creating the db.json file that contains the ligand parameters rather than the charge calculation?